src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/volume/AzureVolumePlugin.java

```diff
@@ -94,10 +100,14 @@ public void deleteInstance(VolumeOrder volumeOrder, AzureUser azureUser) throws
         LOGGER.info(String.format(Messages.Info.DELETING_INSTANCE_S, volumeOrder.getInstanceId()));
         Azure azure = AzureClientCacheManager.getAzure(azureUser);
         String resourceName = AzureGeneralUtil.defineResourceName(volumeOrder.getInstanceId());
-        String subscriptionId = azureUser.getSubscriptionId();
-        String resourceId = buildResourceId(subscriptionId, resourceName);
         
-        doDeleteInstance(azure, resourceId); 
+        if (AzureResourceGroupOperationUtil.existsResourceGroup(azure, resourceName)) {
+            doDeleteResourceGroup(azure, resourceName);
```

Comment:
Will this method repeat in other plugins? Maybe you can use an auxiliary class.

Reply:
This point is decisive in which option should be followed at the time of exclusion, the first option is obvious to remove via resource group but otherwise, it will always change according to each plugin. If you have a better suggestion please share.